app/controllers/admin/custom_fields_controller.rb

```diff
@@ -234,6 +242,26 @@ def update_location
     end
   end
 
+  def update_expiration
+    listing_expiration_enabled = params[:listing_expiration_enabled] == "enabled"
+
+    success = @current_community.update_attributes(
+      { hide_expiration_date: !listing_expiration_enabled })
+
+    if success
+      redirect_to admin_custom_fields_path
+    else
+      flash[:error] = "Expiration field editing failed"
```

Comment:
I was lazy and decided to not add translation, since this shouldn't happen in normal situations.